pages/docs/concepts/asyncapi-document/tags.md

```diff
@@ -0,0 +1,249 @@
+---
```

Comment:
Excellent first draft with growth room for minor improvements, @TRohit20! 

You're so close but still have a few further improvements to make. 😄 **FYI, remember we observe USA (American) spelling, so remember to conduct another review of your spelling in this doc.**

----  

1. **Introduction:**

>  "A `tag` is a label or category that helps you logically group related objects such as channels or servers within an event-driven system. The `tags` object provides a structured way to group channels, operations, or other components based on their functionality, purpose, and/or any other relevant criteria."

➡️ Specify right away that this guide focuses on using tags **in the AsyncAPI specification** to avoid potential confusion, instead of including a mention of _"event-driven system."_ 

2. **Under 'Tags in AsyncAPI Document':**

> "The behaviour and impact of the `tags` object within the AsyncAPI document can vary depending on the context of the object in which it is being used." 

Great sentence, but it would be helpful to list more context where `tags` could be used upfront, instead of waiting to go into detail about each context in their respective subsections.

3. **Under 'Tags in Info Object':**

>   "Tags when defined within the `tags` property of the `info` object provide a high-level categorization and classification for the entire AsyncAPI document."

Let's simplify this sentence. For example, _"When defined within the `tags` property of the `info` object, tags provide a high-level categorization and classification for the entire AsyncAPI document."_

4. **Under 'Tags in Servers Object':**

Clarify if `tags` are a standard part of the AsyncAPI specification or if they are an _optional_ addition. 

5. **Under 'Tags in Channels object', 'Tags in Operations Object', 'Tags in message object':**

Well-written sections, but it would be useful to clarify if tags can be used to filter or search through channels, operations, or messages in some way. If so, could you explain how to do so?

6. **Consistency:**

Ensure consistency in using the back-tick for code keywords. In some places, it is missing... example, in _"tags in servers object,"_ It should be "`tags` in `servers` object".

Reply:
Roger that, Will work on all the improvements suggested along with my edit, I am making changes to the doc per @derberg 's suggestions, will ensure to implement the improvements along. Thank you!😅